new authors and gemini article (#265)
deepset-ai/haystack-home: content/blog/gemini-models-with-google-vertex-for-haystack/index.md

---
layout: blog-post
title: Gemini Models with Google Vertex AI Integration for Haystack 

> description: Multi-Modal and components and function calling with the new Gemini integrations for Haystack

Author: ```suggestion
description: Multi-Modal components and function calling with the new Gemini integrations for Haystack
```

featured_image: thumbnail.jpg
images: ["blog/gemini-models-with-google-vertex-for-haystack/thumbnail.jpg"]

> featured_image_caption: An image or a robot looking at pictures of other robots

Author: ```suggestion
featured_image_caption: An image of a robot looking at pictures of other robots (generated with DALL-E)
```

                    "fahrenheit",
                ],
            },
        },
        "required": ["location"],
    },
)
tool = Tool([get_current_weather_func])

```

We can use this tool with the `GeminiChatGenerator` and ask it to tell us how the function should be called to answer the question “What is the temperature in celsius in Berlin?”:

```python
from google_vertex_haystack.generators.chat.gemini import GeminiChatGenerator
from haystack.dataclasses import ChatMessage

gemini_chat = GeminiChatGenerator(model="gemini-pro", project_id='YOUR-GCP-PROJECT-ID', tools=[tool])

messages = [ChatMessage.from_user(content = "What is the temperature in celsius in Berlin?")]
res = gemini_chat.run(messages=messages)
res["replies"]

```

With the response we get from this interaction, we can call the function `get_current_weather` and proceed with our chat:

```python
weather = get_current_weather(**res["replies"][0].content)
messages += res["replies"] + [ChatMessage.from_function(content=weather, name="get_current_weather")]

res = gemini_chat.run(messages = messages)
res["replies"][0].content

```

## Building a Full Retrieval-Augmented Generative Pipeline

> Alongside the individual use of the new Gemini components above, you can of course also use them in full [Haystack pipelines](https://docs.haystack.deepset.ai/v2.0/docs/pipelines). Here is an example of a RAG pipeline that does question-answering on webpages using the `[LinkContentFetcher](<https://docs.haystack.deepset.ai/v2.0/docs/linkcontentfetcher>)` and the `GeminiGenerator` using the `gemini-pro-vision` model 👇

Author: ```suggestion
Alongside the individual use of the new Gemini components above, you can of course also use them in full [Haystack pipelines](https://docs.haystack.deepset.ai/v2.0/docs/pipelines). Here is an example of a RAG pipeline that does question-answering on webpages using the [`LinkContentFetcher`](https://docs.haystack.deepset.ai/v2.0/docs/linkcontentfetcher) and the `GeminiGenerator` using the `gemini-pro-vision` model 👇
```
